packages/vm/src/runTx.ts

```diff
@@ -300,6 +300,11 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   let fromAccount = await state.getAccount(caller)
   const { nonce, balance } = fromAccount
 
+  // EIP-3607: Reject transactions from senders with deployed code
+  if (fromAccount.codeHash.toString('hex') !== KECCAK256_NULL_S) {
```

Comment:
last nit, can compare the buffers directly here
`if (!fromAccount.codeHash.equals(KECCAK256_NULL)) {`